### DISCUSSION TITLE
Do you really need a custom block? Exploring alternatives to block development in WordPress

### DISCUSSION THREAD

**Discussion by ndiego** [WordPress/developer-blog-content#278] [Topic Ideas]
Custom block development is powerful, but often you don’t need to go through the hassle of building a custom block to achieve your goals.

This article will explore 4-5 scenarios where alternatives to custom blocks might be more effective. Often, this will be in the form of block patterns. These non-block alternatives will feature one or more of the following tools:

- Content-only editing
- Block naming
- Block locking
- `allowedBlocks`
- Block bindings
- Block variations
- Patterns
- And more

The goal is to showcase the flexibility and power of the pattern "toolset" and other non-block alternatives in WordPress. Each example will link to further resources, both on the Dev Blog and in the documentation, where readers can learn more about each tool.

Additionally, the Block Editor Handbook will be updated to ensure that each tool covered in this post is thoroughly documented.

_(The title is a work in progress)_

**Comment by bph** [comment]
Topic approved, please continue conversation on the issue
- #283

**Comment by ndiego** [comment]
This article has been published: https://developer.wordpress.org/news/2024/08/15/how-to-extend-a-wordpress-block/
